### PR TITLE
Add training code  for gencast

### PR DESCRIPTION
**Pull request by liaoxin2**
### PR changes
 One of APIs

### Describe
Add training code  for gencast

### CODE REVIEW

File: docs/zh/examples/gencast.md [hunk 1]
```diff
@@ -5,6 +5,16 @@
 - 下载目录`dm_graphcast/gencast/stats`下的所有文件放入`./data/stats/`目录下。
 - 下载目录`dm_graphcast/gencast/dataset`下的任意或所有文件（例如：source-era5_date-2019-03-29_res-1.0_levels-13_steps-12.nc）放入`./data/dataset/`目录下。
 
+=== "模型训练命令"
+
+    ``` sh
+    # 设置路径到 PaddleScience/jointContribution 文件夹
+    cd PaddleScience/jointContribution
+    export PYTHONPATH=$PWD:$PYTHONPAT
```
Comment (HydrogenSulfate) [inline docs/zh/examples/gencast.md]:
```suggestion
    export PYTHONPATH=$PWD:$PYTHONPATH
```

File: jointContribution/gencast/denoiser.py [hunk 2]
```diff
@@ -66,12 +67,16 @@ def __init__(
...
         layers = []
         input_size = 2 * num_frequencies
         num_layers = len(output_sizes)
         for i, output_size in enumerate(output_sizes):
-            linear_layer = nn.Linear(input_size, output_size)
+            limit = math.sqrt(6 / input_size)
+            weight_attr = paddle.framework.ParamAttr(
+                initializer=paddle.nn.initializer.Uniform(low=-limit, high=limit)
+            )
+            linear_layer = nn.Linear(input_size, output_size, weight_attr=weight_attr)
```
Comment (HydrogenSulfate) [inline jointContribution/gencast/denoiser.py:79]: 可以直接用 ppsci.utils.initializer 这个模块下的API吧，这样不用手算初始化的参数分布

File: jointContribution/gencast/run_gencast.py [hunk 3]
```diff
@@ -26,20 +26,88 @@
...
-        raise ValueError("predictions must have dim 'sample' with size at least 2.")
-    sum_dims = ["sample", "sample2"]
-    preds2 = predictions.rename({"sample": "sample2"})
-    num_samps = predictions.sizes["sample"]
-    num_samps2 = (num_samps - 1) if bias_corrected else num_samps
-    mean_abs_diff = np.abs(predictions - preds2).sum(dim=sum_dims, skipna=False) / (
-        num_samps * num_samps2
+class CustomDataLoader(paddle.io.Dataset):
+    def __init__(self, target_lead_times, cfg):
+        super(CustomDataLoader, self).__init__()
```
Comment (HydrogenSulfate) [inline jointContribution/gencast/run_gencast.py]:
```suggestion
        super().__init__()
```

**Review by HydrogenSulfate — COMMENTED** [review]
LGTM

**Review by HydrogenSulfate — APPROVED** [review]
